### DISCUSSION TITLE
Composability with mount points

### DISCUSSION THREAD

**Discussion by derrickburns** [fluxcd/flux2#1702] [Proposals]
There are many ways to build k8s YAML manifests: helm templating, jsonnet, kustomizations, etc. The problem with all of them is that that are not mutually composable.  

I have occasionally come across a helm chart that lacked the templating I needed. However, in my custom system I created a way to sequence transformations on the output of helm.  

More generally, what if we had a general dependency tree.   What if you had for a dependency an associated mount point. Evaluating a dependency means reifying the manifests as files that we mount under a directory associated with the dependency.  Then, the output of the dependency would be available as the input for the next step. 

Secret decryption could be implemented this way.  In this re-imagination, a compliant controller would publish its output to a new controller that would coordinate the workflow.   This allows new controllers to be added without extending the Flux API.

The final sink of that graph of dependencies is the K8s api server itself.

**Comment by stefanprodan** [comment]
> I have occasionally come across a helm chart that lacked the templating I needed.

Flux does support customizing the helm output, please see the docs https://fluxcd.io/docs/components/helm/helmreleases/#post-renderers
